pkg/scaler/prober.go

```diff
@@ -68,50 +74,64 @@ type probeResult struct {
 	resultRun int32
 }
 
-func (p *prober) getClientFromSecret(secretName string) (kubernetes.Interface, error) {
+// getClientFromSecret constructs a Kubernetes client based on the supplied secret and
+// return it along with the SHA256 checksum of the kubeconfig in the secret
+// but only if the SHA256 checksum of the kubeconfig in the secret differs from oldSHA.
+func (p *prober) getClientFromSecret(secretName string, oldSHA []byte) (kubernetes.Interface, []byte, error) {
```

Comment:
When is this created. Why do we need this logic to avoid creating a k8s client if the secret didn't change?